diff -up ./mm/huge_memory.c.pmd ./mm/huge_memory.c
--- ./mm/huge_memory.c.pmd	2018-11-02 11:12:17.000000000 +0900
+++ ./mm/huge_memory.c	2018-11-02 11:32:28.000000000 +0900
@@ -1370,7 +1370,7 @@ static int do_huge_pmd_wp_page_fallback(
 	unsigned long haddr = address & HPAGE_PMD_MASK;
 	spinlock_t *ptl;
 	pgtable_t pgtable;
-	pmd_t _pmd = {{{0}}};
+	pmd_t _pmd = {0};
 	int ret = 0, i;
 	struct page **pages;
 	unsigned long mmun_start;	/* For mmu_notifiers */
@@ -2313,7 +2313,7 @@ static int __split_huge_page_map(struct
 {
 	struct mm_struct *mm = vma->vm_mm;
 	spinlock_t *ptl;
-	pmd_t *pmd, _pmd = {{{0}}};
+	pmd_t *pmd, _pmd = {0};
 	int ret = 0, i;
 	pgtable_t pgtable;
 	unsigned long haddr;
