diff -up ./arch/x86/kernel/vm86_32.c.tlb ./arch/x86/kernel/vm86_32.c
--- ./arch/x86/kernel/vm86_32.c.tlb	2018-03-22 06:40:12.000000000 +0900
+++ ./arch/x86/kernel/vm86_32.c	2018-04-15 22:46:42.000000000 +0900
@@ -194,7 +194,7 @@ static void mark_screen_rdonly(struct mm
 	pte_unmap_unlock(pte, ptl);
 out:
 	up_write(&mm->mmap_sem);
-	flush_tlb();
+	flush_tlb_mm_range(mm, 0xA0000, 0xA0000 + 32*PAGE_SIZE, 0UL);
 }
 
 
