diff -aurp a/arch/x86/kernel/cpu/mcheck/mce.c b/arch/x86/kernel/cpu/mcheck/mce.c
--- a/arch/x86/kernel/cpu/mcheck/mce.c	2019-10-05 05:27:35.000000000 -0300
+++ b/arch/x86/kernel/cpu/mcheck/mce.c	2020-03-30 12:06:36.546549480 -0300
@@ -52,6 +52,13 @@
 
 #include "mce-internal.h"
 
+#ifndef set_mce_nospec
+static inline int set_mce_nospec(unsigned long pfn)
+{
+	return 0;
+}
+#endif
+
 static DEFINE_MUTEX(mce_chrdev_read_mutex);
 
 #define mce_log_get_idx_check(p) \
diff -aurp a/drivers/scsi/megaraid/megaraid_sas_debugfs.c b/drivers/scsi/megaraid/megaraid_sas_debugfs.c
--- a/drivers/scsi/megaraid/megaraid_sas_debugfs.c	2019-10-05 05:27:35.000000000 -0300
+++ b/drivers/scsi/megaraid/megaraid_sas_debugfs.c	2020-03-30 13:20:30.225702075 -0300
@@ -28,6 +28,7 @@
 #include <linux/kernel.h>
 #include <linux/types.h>
 #include <linux/pci.h>
+#include <linux/uio.h>
 #include <linux/interrupt.h>
 #include <linux/compat.h>
 #include <linux/irq_poll.h>
